File: mobile/lib/models/order.dart
class Order {
  final int id;
  final String orderCode;
  final String type;
  final String status;
  final String? createdAt;
  final double subtotal;
  final double discountAmount;
  final double taxAmount;
  final double finalAmount;
  final String? paymentMethod;
  final String paymentStatus;
  final double? cashReceived;
  final double? changeAmount;
  final String? paidAt;
  final TableInfo? table;
  final CustomerInfo? customer;
  final List<OrderItem> items;

  Order({
    required this.id,
    required this.orderCode,
    required this.type,
    required this.status,
    this.createdAt,
    required this.subtotal,
    required this.discountAmount,
    required this.taxAmount,
    required this.finalAmount,
    this.paymentMethod,
    required this.paymentStatus,
    this.cashReceived,
    this.changeAmount,
    this.paidAt,
    this.table,
    this.customer,
    required this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] as int,
      orderCode: json['order_code'] as String,
      type: json['type'] as String,
      status: json['status'] as String,
      createdAt: json['created_at'] as String?,
      subtotal: (json['subtotal'] as num).toDouble(),
      discountAmount: (json['discount_amount'] as num?)?.toDouble() ?? 0,
      taxAmount: (json['tax_amount'] as num?)?.toDouble() ?? 0,
      finalAmount: (json['final_amount'] as num).toDouble(),
      paymentMethod: json['payment_method'] as String?,
      paymentStatus: json['payment_status'] as String,
      cashReceived: (json['cash_received'] as num?)?.toDouble(),
      changeAmount: (json['change_amount'] as num?)?.toDouble(),
      paidAt: json['paid_at'] as String?,
      table: json['table'] != null
          ? TableInfo.fromJson(json['table'] as Map<String, dynamic>)
          : null,
      customer: json['customer'] != null
          ? CustomerInfo.fromJson(json['customer'] as Map<String, dynamic>)
          : null,
      items: (json['items'] as List<dynamic>?)
              ?.map((e) => OrderItem.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }
}

class TableInfo {
  final int id;
  final String name;

  TableInfo({required this.id, required this.name});

  factory TableInfo.fromJson(Map<String, dynamic> json) {
    return TableInfo(
      id: json['id'] as int,
      name: json['name'] as String,
    );
  }
}

class CustomerInfo {
  final int id;
  final String name;
  final String? email;
  final String? phone;

  CustomerInfo({
    required this.id,
    required this.name,
    this.email,
    this.phone,
  });

  factory CustomerInfo.fromJson(Map<String, dynamic> json) {
    return CustomerInfo(
      id: json['id'] as int,
      name: json['name'] as String,
      email: json['email'] as String?,
      phone: json['phone'] as String?,
    );
  }
}

class OrderItem {
  final int productId;
  final String productName;
  final double quantity;
  final double unitPrice;
  final double subtotal;
  final String? notes;

  OrderItem({
    required this.productId,
    required this.productName,
    required this.quantity,
    required this.unitPrice,
    required this.subtotal,
    this.notes,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      productId: json['product_id'] as int,
      productName: json['product_name'] as String,
      quantity: (json['quantity'] as num).toDouble(),
      unitPrice: (json['unit_price'] as num).toDouble(),
      subtotal: (json['subtotal'] as num).toDouble(),
      notes: json['notes'] as String?,
    );
  }
}

File: app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

#[Fillable([
    'store_id', 'shift_id', 'table_id', 'customer_id', 'cashier_id', 'promo_id', 'order_code', 'type', 'status', 'notes',
    'subtotal', 'discount_amount', 'tax_rate', 'tax_amount', 'final_amount',
    'payment_method', 'payment_status', 'cash_received', 'change_amount', 'paid_at',
    'created_at', 'completed_at', 'cancelled_at', 'cancellation_reason',
])]
class Order extends Model
{
    use SoftDeletes;

    public function store(): BelongsTo
    {
        return $this->belongsTo(Store::class);
    }

    public function shift(): BelongsTo
    {
        return $this->belongsTo(CashierShift::class, 'shift_id');
    }

    public function table(): BelongsTo
    {
        return $this->belongsTo(DiningTable::class, 'table_id');
    }

    public function customer(): BelongsTo
    {
        return $this->belongsTo(Customer::class);
    }

    public function cashier(): BelongsTo
    {
        return $this->belongsTo(User::class, 'cashier_id');
    }

    public function promo(): BelongsTo
    {
        return $this->belongsTo(Promo::class);
    }

    public function items(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    public function feedback(): \Illuminate\Database\Eloquent\Relations\HasOne
    {
        return $this->hasOne(OrderFeedback::class);
    }

    public function grossProfit(): float
    {
        return (float) $this->items->sum(fn ($item) =>
            ($item->unit_price - $item->buy_price) * $item->quantity
        );
    }

    protected function casts(): array
    {
        return [
            'subtotal'            => 'decimal:2',
            'discount_amount'     => 'decimal:2',
            'tax_rate'            => 'decimal:2',
            'tax_amount'          => 'decimal:2',
            'final_amount'        => 'decimal:2',
            'cash_received'       => 'decimal:2',
            'change_amount'       => 'decimal:2',
            'paid_at'             => 'datetime',
            'completed_at'        => 'datetime',
            'cancelled_at'        => 'datetime',
        ];
    }
}
